sdk/daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala

```diff
@@ -332,59 +368,91 @@ trait LongTests { this: UpgradesSpec =>
     }
 
     s"Succeeds when v1 upgrades to v2 and then v3 ($suffix)" in {
-      testPackagePair(
-        "test-common/upgrades-SuccessUpgradingV2ThenV3-v1.dar",
-        "test-common/upgrades-SuccessUpgradingV2ThenV3-v2.dar",
-        assertPackageUpgradeCheck(None),
-      )
-      testPackagePair(
-        "test-common/upgrades-SuccessUpgradingV2ThenV3-v1.dar",
-        "test-common/upgrades-SuccessUpgradingV2ThenV3-v3.dar",
-        assertPackageUpgradeCheck(None),
-      )
+      for {
+        v1Upload <- uploadPackage("test-common/upgrades-SuccessUpgradingV2ThenV3-v1.dar")
+        v2Upload <- uploadPackage("test-common/upgrades-SuccessUpgradingV2ThenV3-v2.dar")
+        v3Upload <- uploadPackage("test-common/upgrades-SuccessUpgradingV2ThenV3-v3.dar")
+      } yield {
+        val cantonLog = Source.fromFile(s"$cantonTmpDir/canton.log")
```

Comment:
In Java, we would write something like:
```
try (cantonLog = Source.fromFile(s"$cantonTmpDir/canton.log")) {
 // etc.
}
```
and have it manage the file resource for us. Sadly, this isn't directly possible in Scala.

However, I believe we can write:
```
try (Source cantonLog = Source.fromFile(s"$cantonTmpDir/canton.log")) {
  // etc.
}
```
or:
```
import scala.util.Using
Using(Source.fromFile(s"$cantonTmpDir/canton.log")) { cantonLog =>
  // etc.
}
```

Reply:
I've opted to read the file entirely via `Using` before trying any assertions over it, like

```
...
        v3Upload <- uploadPackage("test-common/upgrades-SuccessUpgradingV3ThenV2-v3.dar")
        v2Upload <- uploadPackage("test-common/upgrades-SuccessUpgradingV3ThenV2-v2.dar")
        rawCantonLog <- Future.fromTry(Using(Source.fromFile(s"$cantonTmpDir/canton.log"))(_.mkString))
      } yield {
        forEvery(
...
```